modal/_container_entrypoint.py

```diff
@@ -569,9 +597,11 @@ def breakpoint_wrapper():
             post_snapshot_methods = _find_callables_for_obj(imp_fun.obj, _PartialFunctionFlags.ENTER_POST_SNAPSHOT)
             call_lifecycle_functions(event_loop, container_io_manager, list(post_snapshot_methods.values()))
 
+        finalized_function = finalize_function(imp_fun, container_io_manager)
```

Comment:
if there's an exception here, will this be handled properly? iirc there's a context manager to handle user exceptions that we might want to user here

although the exception flow is i guess an existing problem, so it's not something we have to solve in this PR